instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+
+class CommentSection extends React.Component {
```

Comment:
So this is cool, I like that you were able to come up with this and do it. However, you're going to have a split in your state. You'll have one set of comments on App-state and another on CommentSection-state, and they're going to diverge. Instead, if you want to keep state here, that's fine, but then only use it for `this.state.comment` and retain the method `commentHandler`. Then move your `submitNewComment` method to your App component and pass it down, and change the logic so that it sets the state on App appropriately. Then, it will re-render everything and drip down so that you'll have a SSOT (single source of truth) on App state. Does this make sense?